test/extended/router/gatewayapicontroller.go

```diff
@@ -246,6 +249,90 @@ var _ = g.Describe("[sig-network-edge][OCPFeatureGate:GatewayAPIController][Feat
 		g.By("Validating the http connectivity to the backend application")
 		assertHttpRouteConnection(defaultRoutename)
 	})
+
+	g.It("Ensure OSSM subscription, istiod deployment and the istio could be deleted and then get recreated [Serial]", func() {
+		const (
+			operatorNamespace      = "openshift-operators"
+			ingressNamespace       = "openshift-ingress"
+			ossmSubscriptionName   = "servicemeshoperator3"
+			csvName                = "servicemeshoperator3.v3.0.0"
+			ossmOperatorDeployment = "servicemesh-operator3"
+			istiodDeployment       = "istiod-openshift-gateway"
+			istioName              = "openshift-gateway"
+		)
+
+		// deleted the OSSM subscription and then checked if it was restored
+		g.By(fmt.Sprintf("Try to delete the subscription %s", ossmSubscriptionName))
+		_, err := oc.AsAdmin().WithoutNamespace().Run("delete").Args("-n", operatorNamespace, "subscription/"+ossmSubscriptionName).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By(fmt.Sprintf("Wait until the the OSSM subscription %s is automatically created successfully", ossmSubscriptionName))
+		pollWaitSubscriptionCreated(oc, operatorNamespace, ossmSubscriptionName)
+
+		g.By(fmt.Sprintf("Wait until the the OSSM csv %s is automatically created successfully", csvName))
+		pollWaitOssmCsvCreated(oc, operatorNamespace, csvName)
+
+		// deleted the istiod deployment and then checked if it was restored
+		deleteDeploymentAndWaitAvailableAgain(oc, istiodDeployment, ingressNamespace)
```

Comment:
Can we be consistent in the way we do "delete and wait"? I mean for the subscription and Istio CR you delete with `oc.AsAdmin()` while here the deletion is hidden in the helper function.

Reply:
I had used deleteDeploymentAndWaitAvailableAgain function to delete 3 deployments. I will spit this function and update it to be consistent  as the before test steps. lt looked like below:
```
                g.By(fmt.Sprintf("Try to delete the istiod deployment in %s namespace", ingressNamespace))
                err = oc.AdminKubeClient().AppsV1().Deployments(ingressNamespace).Delete(context.Background(), istiodDeployment, metav1.DeleteOptions{})
                o.Expect(err).NotTo(o.HaveOccurred())
        
                g.By(fmt.Sprintf("Wait until the istiod deployment in %s namespace is automatically created successfully", ingressNamespace))
                pollWaitDeploymentCreated(oc, ingressNamespace, istiodDeployment)
```